### PR TITLE
QL: add ql-for-ql query for detecting bad predicate qldoc

### PR DESCRIPTION
**Pull request by erik-krogh**
Inspired by some qldoc from https://github.com/github/codeql/pull/8014 

[Here is the style guide](https://github.com/github/codeql/blob/main/docs/qldoc-style-guide.md#qldoc-for-predicates). 

And drive-by fix most of the instances I found throughout JS.

### CODE REVIEW

**Review by kaeluka — COMMENTED** [review]
Thank you. This makes sense, and I only have some stylistic comments.

I have also browsed the open code scanning warnings, and fixing the warnings to satisfy this query would increase consistency of documentation and not be (too) harmful in almost all cases 👍 

Further questions:

 * What is the reason to leave out class and module comments?
 * PRs like these open a lot of warnings that remain in the code base. Do we have a strategy to clean those up in the future? By following the [Boy Scout Rule](https://stevenpcurtis.medium.com/the-boy-scout-rule-in-coding-a0b4d8130798), I'm assuming?

**Review by annarailton — APPROVED** [review]
Happy with the changes made to the two `javascript/ql/experimental/adaptivethreatmodeling` files touched by this PR.

File: ql/ql/src/queries/style/docs/PredicateDocs.ql [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import ql
+
+string docLines(Predicate pred) {
+  result =
+    pred.getQLDoc().getContents().replaceAll("/**", "").replaceAll("*", "").splitAt("\n").trim()
+}
+
+from Predicate pred, string message
+where
+  // only considering qldocs that look like a class-doc, to avoid reporting way too much.
```
[the remark this answers is not in the document]
Comment (erik-krogh) [reply ql/ql/src/queries/style/docs/PredicateDocs.ql:20]: Not necessarily. There were about 2500 warnings without that restriction, which is way too much.  
And some of those alerts were kinda OK.  

But I generally like alerts produced with this restriction, and they seem to be easily fixable.  
We might widen it in the future, but I have no such plans.

**Comment by erik-krogh** [comment]
> What is the reason to leave out class and module comments?

We already got another query for class qldoc, and module comments are so loose that I don't think we can flag those. 

> PRs like these open a lot of warnings that remain in the code base. Do we have a strategy to clean those up in the future? By following the [Boy Scout Rule](https://stevenpcurtis.medium.com/the-boy-scout-rule-in-coding-a0b4d8130798), I'm assuming?

Yes. People tend to be good at fixing issues that are presented to them by code-scanning, so I don't expect the alert-count to increase (much).  
And a few languages have made an effort to decrease the number of alerts, which is nice.

File: ql/ql/src/queries/style/docs/PredicateDocs.ql (hunk 1 above)
[the remark this answers is not in the document]
Comment (kaeluka) [reply ql/ql/src/queries/style/docs/PredicateDocs.ql:20]: > But I generally like alerts produced with this restriction, and they seem to be easily fixable.
We might widen it in the future, but I have no such plans.

Ah, that's where this specific restriction is coming from. It seemed a bit arbitrary to focus on specifically *those* comments. Thx for clarifying.

**Review by kaeluka — DISMISSED** [review]
LGTM 👍

File: ql/ql/src/queries/style/docs/PredicateDocs.ql [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  docLines(pred).matches(["A", "An", "The"] + " %") and // looks like a class doc.
```
Comment (tausbn) [inline ql/ql/src/queries/style/docs/PredicateDocs.ql:21]: Should we perhaps exclude `private` predicates? Of course ideally we would want to be consistent in our usage, but in practice we're a bit more loose when it comes to predicates that are not visible externally.

**Review by tausbn — APPROVED** [review]
Looks good to me! :+1:

**Pull request merged**